core/src/com/unciv/models/ruleset/Building.kt

```diff
@@ -200,6 +197,147 @@ class Building : NamedStats(), IConstruction {
+            textList += FormattedLine()
+        if (requiredTech != null)
+            textList += FormattedLine("Required tech: [$requiredTech]",
+                link="Technology/$requiredTech")
+        if (requiredBuilding != null)
+            textList += FormattedLine("Requires [$requiredBuilding] to be built in the city",
+                link="Building/$requiredBuilding")
+        if (requiredBuildingInAllCities != null)
+            textList += FormattedLine("Requires [$requiredBuildingInAllCities] to be built in all cities",
+                link="Building/$requiredBuildingInAllCities")
+
+        val resourceRequirements = getResourceRequirements()
+        if (resourceRequirements.isNotEmpty()) {
+            textList += FormattedLine()
+            for ((resource, amount) in resourceRequirements) {
+                textList += FormattedLine(
+                    // the 1 variant should deprecate some time
+                    if (amount == 1) "Consumes 1 [$resource]" else "Consumes [$amount] [$resource]",
+                    link="Resources/$resource", color="#F42" )
+            }
+        }
+
+        if (providesFreeBuilding != null) {
+            textList += FormattedLine()
+            textList += FormattedLine("Provides a free [$providesFreeBuilding] in the city",
+                link="Building/$providesFreeBuilding")
+        }
+
+        val stats = this.clone()
+        val percentStats = getStatPercentageBonuses(null)
+        val specialists = newSpecialists()
+        if (uniques.isNotEmpty() || !stats.isEmpty() || !percentStats.isEmpty() || this.greatPersonPoints != null || specialists.isNotEmpty())
+            textList += FormattedLine()
+
+        if (uniques.isNotEmpty()) {
+            if (replacementTextForUniques.isNotEmpty())
+                textList += FormattedLine(replacementTextForUniques)
+            else
+                for (unique in getUniquesStrings()) textList += FormattedLine(unique)
+        }
+
+        if (!stats.isEmpty()) {
+            textList += FormattedLine(stats.toString())
+        }
+
+        if (!percentStats.isEmpty()) {
+            if (percentStats.production != 0f)
+                textList += FormattedLine(percentStats.production.formatSignedInt() + "% {Production}")
+            if (percentStats.gold != 0f)
+                textList += FormattedLine(percentStats.gold.formatSignedInt() + "% {Gold}")
+            if (percentStats.science != 0f)
+                textList += FormattedLine(percentStats.science.formatSignedInt() + "% {Science}")
+            if (percentStats.food != 0f)
+                textList += FormattedLine(percentStats.food.formatSignedInt() + "% {Food}")
+            if (percentStats.culture != 0f)
+                textList += FormattedLine(percentStats.culture.formatSignedInt() + "% {Culture}")
+        }
+
+        if (this.greatPersonPoints != null) {
+            val gpp = this.greatPersonPoints!!
+            if (gpp.production != 0f) textList +=
```

Comment:
Here too, you can use the 'stat to great person' mapping although I hate it

Reply:
Both are more or less copies of the 'old' getDescription implementation - had I optimized everything, well, the diff engine might not have found anything left to sync on...

Reply:
I just couldn't use it as is. (#civs + #overview uses + #city screens opened) instances in a run when 1 is enough? See next commit...